numpy/random/mtrand/mtrand.pyx

```diff
@@ -158,7 +158,9 @@ cdef object cont0_array(rk_state *state, rk_cont0 func, object size,
     cdef npy_intp i
 
     if size is None:
-        return func(state)
+        with lock, nogil:
+            rv = func(state)
```

Comment:
Any reason not to just do

    with lock, nogil:
        return func(state)

?